### DISCUSSION TITLE
mysql cdc 报 unexpected block data

### DISCUSSION THREAD

**Discussion by zdf321** [apache/flink-cdc#644] [Q&A]
mysql cdc用的包是flink-sql-connector-mysql-cdc-2.1.0.jar，flink版本是1.13.3，同时mysql开启了binlog
在sql-client.sh中执行如下SQL：
CREATE TABLE products (
    id INT,
    name STRING,
    description STRING,
    PRIMARY KEY (id) NOT ENFORCED
  ) WITH (
    'connector' = 'mysql-cdc',
    'hostname' = 'x.x.x.x',
    'port' = '3306',
    'username' = 'root',
    'password' = '123456',
    'database-name' = 'test',
    'table-name' = 'products'
  );

select * from products;

报错如下：
[ERROR] Could not execute SQL statement. Reason:
java.io.StreamCorruptedException: unexpected block data

**Comment by cuoben** [comment]
类加载顺序问题，flink默认是child-first，在flink的flink-conf.yaml文件中添加`classloader.resolve-order: parent-first` 改成parent-first，重启集群即可。

**Comment by yqwoe** [comment]
1.14.5 mysql-cdc   一样的问题，

**Comment by tianhao960** [comment]
same issue. flink 1.14.5 cdc 2.2.1

**Comment by sleten** [comment]
直接把cdc的jar包放到lib下后不重启flink集群，进入sql命令行会报这个错，重启flink集群再进sql命令行就好了
flink：1.13.5
cdc：2.2.0

**Reply by sleten** [reply]
If you directly put the jar package of cdc under lib without restarting the flink cluster, this error will be reported when you enter the sql command line, restart the flink cluster and then enter the sql command line again
flink：1.13.5
cdc：2.2.0

**Comment by zhiqiang-dev** [comment]
这个错误主要是切换了类加载器，然后用这个类加载器做反序列化，如果类的版本不一致就可能报这个错误。
看一下 tm jm 是否有日志，然后确定一下具体是哪里报错。可以远程 debug 一下。 我把 cdc 自己打包的源码和 Flink 对应的版本都调整成一致就可以了。
确定 scala 版本，
确定 flink 版本
确定 flink-cdc 打包的版本都一致。

这个错误一般在 jm 的异常里面可以看到
1.16.1 主要是这行代码的问题
<img width="1093" alt="image" src="https://github.com/ververica/flink-cdc-connectors/assets/16716138/62ea91b9-af39-4140-a372-7f9f7e8eb126">

**Reply by zhiqiang-dev** [reply]
classloader.resolve-order: child-first 也没什么问题
flink 1.16.1 +  cdc 2.3 release 测试可以。

**Reply by zhiqiang-dev** [reply]
中间测试的时候用 stop-cluster 停止集群的时候，没有彻底停止。更新 jar 后需要检查一下。 
如果 pid 没变的话，用 jps 查看一下，然后 kill 后，再重启。
